### DISCUSSION TITLE
Having trouble publishing to npm

### DISCUSSION THREAD

**Discussion by shairez** [bikecoders/ngx-deploy-npm#360] [Q&A] [answered]
Hi! 
I've been fighting with this for a few days, trying all sorts of possible solutions, but just can't seem to pass the damn AUTH error lol

Here's the issue
https://github.com/hirezio/lembas/runs/7928098387?check_suite_focus=true#step:4:72

I run only "publish" without the "semver" for now just to stop it from increasing versions unnecessarily. 

It keeps ignoring my environment variable for NODE_AUTH_TOKEN although I have an almost identical setup to the "jscutlery/semver" library (I based it on their github action files, but I'm just using npm instead of yarn)

If you have any clues as to what is happening here and where is a potential lead to follow in my investigation, you'll have my eternal gratitude 😊

Thanks in advance! 🙏

**Comment by dianjuar** [comment]
I really think it's because you are using _yarn_ instead of_npm_

I faced the same error before. Somewhere here on ngx-deploy-npm 's actions there is the warning. 

I can provide links in the future to support my statement but I'm 95% secure you solve  the problem using _npm_

**Reply by shairez** [reply]
Thanks Diego!

But I don't recall using yarn

Can you point me to where you saw I'm using yarn?

Thanks!

**Reply by dianjuar** [reply]
It was nice to see you at the ng-conf.
Glad that destiny put us together in the same place and we could see each other. 

As I told you on the ng-conf, there were rough weeks in the middle of a journey, and it was hard for me to give you the attention I wanted to dedicate.

I realize that my comment has no sense.
I will delete it in the next few days; I hope the whole threat goes away to have a clear discussion.

**Comment by dianjuar** [comment]
I would double-check that you have your variable `NPM_TOKEN` in the GitHub secret configuration.

It could seem obvious, but you will be surprised by the number of times a simple space or a misplaced character ruined everything.

**Reply by dianjuar** [reply]
I see that you are using [`npm whoami`](https://github.com/hirezio/lembas/runs/7928098387?check_suite_focus=true#step:4:13) to verify you have the right auth token, and it's working 👍 .

This recommendation could be marked as checked ✅ .

**Comment by dianjuar** [comment]
Something weird is going on.

I see you have [`npm whoami`](https://github.com/hirezio/lembas/runs/7928098387?check_suite_focus=true#step:4:19) to verify the logged user and it's working; you are logged in using `hirez.io` user.
That should be enough to guarantee that you have the proper configuration regarding NPM authentication.

The only difference between whoami step and the publishment step is that you have an extra token, [`NPM_TOKEN`](https://github.com/hirezio/lembas/blob/main/.github/actions/release/action.yml#L25). Here at ngx-deploy-npm, we only use [`NODE_AUTH_TOKEN`](https://github.com/bikecoders/ngx-deploy-npm/blob/main/.github/workflows/publishment.yml#L72), and it's working like a charm.

ngx-deploy-npm uses ngx-deploy-npm to publish to npm (I know, it sounds like builder inception), so I believe `NODE_AUTH_TOKEN` is the correct variable to put your NPM token.

Should we try using only `NODE_AUTH_TOKEN`?

**Reply by shairez** [reply]
Hey brother, just saw this reply now

Yeah, I used just the NODE_AUTO_TOKEN at one point, and just the NPM_TOKEN, and then both

It's really really a weird problem I'm facing

I'll try removing it again, but I don't think this is the issue

**Comment by dianjuar** [comment]
@shairez, Can you share how did you create the NPM token?

**Reply by shairez** [reply]
I generated the token on npmjs.com (with the type of "automation")

**Comment by shairez** [comment, marked as answer]
OK I SOLVED IT! 🎉🎉

It turns out I had misconfigured the registry url

This is what I had:

```
"publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.com"
  }
```

but it should have ended with `.org` 🤦‍♂️

```
    "registry": "https://registry.npmjs.org"
```


Now it works 💪

Thanks for the support @dianjuar 🙏❤

**Reply by dianjuar** [reply]
Glad you finally worked it out 🚀 
That's a hard catch 😮‍💨 

I didn't notice you had that on your `package.json`.
If your intention is to have it as a public package on NPM, you don't need that.
This project is published on NPM and we don't have any of that on the [`package.json`](https://github.com/bikecoders/ngx-deploy-npm/blob/main/packages/ngx-deploy-npm/package.json).

I guess that's why I didn't verify it in the first place.
Maybe you can remove it to avoid this type of issue in the future?

**Reply by shairez** [reply]
thanks @dianjuar !

As the saying goes - "it works.. don't touch it" 😅
